docs/usage.md

```diff
@@ -0,0 +1,212 @@
+
+Sysctl option comparison compares the semantics of the proc handler function and
+semantics of all functions using the data variable that the sysctl option sets.
+
+It is possible to use patterns to specify a number of multiple sysctl options at
+once such as:
+
+- `kernel.*`
+- `kernel.{sysctl-1|sysctl-2}`
+
+Currently, these sysctl option groups are supported: `kernel.*`,
+`vm.*`, `fs.*`, `net.core.*`, `net.ipv4.conf.*`.
+
+#### Options
+
+- `KERNEL_DIR`: Path to a kernel's root directory.
+- `SNAPSHOT_DIR`: Output directory for storing the created snapshot.
+- `SYMBOL_LIST`: Path to a file containing a list of symbols (each symbol on
+  a single line) to be prepared for comparison.
+  In case `--sysctl` is used, the list is interpreted as a list of sysctl
+  parameters.
+- `--sysctl`: Compares sysctl option.
+- `--no-souce-dir`: Does not store the path to the source kernel directory in
+  the snapshot.
+
+### c) `llvm-to-snapshot`: snapshot generation from a single LLVM IR file
+
+```sh
+diffkemp llvm-to-snapshot PROJ_DIR LLVM_FILE SNAPSHOT_DIR SYMBOL_LIST
+```
+
+This command can be used if the project is already compiled into a single
+LLVM IR file.
+The file name is given in `LLVM_FILE` and must be relative to `PROJ_DIR`.
+
+#### Options
+
+- `PROJ_DIR`: Path to a project's root directory.
+- `LLVM_FILE`: Path to the LLVM IR file relative to the project's root
+  directory.
+- `SNAPSHOT_DIR`: Output directory for storing the created snapshot.
+- `SYMBOL_LIST`: Path to a file containing a list of symbols (each symbol on
+  a single line) which should be prepared for comparison.
+
+## 2. Semantic comparison command
+
+```sh
+diffkemp compare SNAPSHOT_DIR_1 SNAPSHOT_DIR_2
+```
+
+This command takes two snapshots and compares them for semantic equality.
+
+Syntactic diffs of the discovered differences are stored in separate files
+(one file for each compared function that is different) in a newly created
+directory. The name of the directory can be specified using the `-o` option,
+otherwise it is generated automatically.
+
+### Options
+
+- `SNAPSHOT_DIR_1`, `SNAPSHOT_DIR_2`: Paths to directories containing snapshots
+  of the project's version to be compared.
+- `--output-dir OUTPUT_DIR`, `-o OUTPUT_DIR`: Name of the output directory.
+- `--custom-patterns CUSTOM_PATTERNS`, `-p CUSTOM_PATTERNS`: Path to a custom
+  pattern file or a configuration.
+- `--show-diff`: Shows/creates syntactic diffs for symbols evaluated as
+  semantically different (used by default, `--no-show-diff` turns this off).
+- `--full-diff`: Shows syntactic diff for all functions (even semantically
+  equivalent ones).
+- `--enable-pattern`, `--disable-pattern`: Enables/disables specified built-in
+  pattern(s).
+- `--enable-all-patterns`,`--disable-all-patterns`: Enables/disables all
+  supported built-in patterns. Be careful with `--enable-all-patterns`, as it
+  also enables the `control-flow-only` pattern, which is not
+  semantic-preserving.
```

Comment:
There are more patterns which are disabled by default. How about this:
```suggestion
- `--enable-all-patterns`,`--disable-all-patterns`: Enables/disables all
  supported built-in patterns. Be careful, `--enable-all-patterns`
  also enables patterns which are not on by default and may not be 
  semantics preserving.
```
Later, we'll add a reference to the documentation of patterns.